lib/improver/wind_calculations/wind_gust_diagnostic.py

```diff
@@ -208,22 +208,15 @@ def process(self, cube_gust, cube_ws):
         ws_time = req_cube_ws.coords('time')
         if len(wg_time) == 0 or len(ws_time) == 0:
             raise ValueError(msg)
-        wg_time = wg_time[0]
-        ws_time = ws_time[0]
-        for i, point in enumerate(wg_time.points):
-            if point != ws_time.points[i]:
-                if wg_time.bounds is None:
-                    raise ValueError(msg)
-                if len(wg_time.bounds) >= i:
-                    if len(wg_time.bounds[i]) == 2:
-                        if ws_time.points[i] < wg_time.bounds[i][0]:
-                            raise ValueError(msg)
-                        elif ws_time.points[i] > wg_time.bounds[i][1]:
-                            raise ValueError(msg)
-                    else:
-                        raise ValueError(msg)
-                else:
-                    raise ValueError(msg)
+
+        if not all(wg_point == ws_point for wg_point, ws_point
+                   in zip(wg_time[0].points, ws_time[0].points)):
+            if wg_time[0].bounds is None:
```

Comment:
Do you think any of these should raise a different error to the "Could not match time coordinate" error? For example here it is raised because the wind gust time coordinate does not have bounds on it, not actually because the points don't match. This might be easier to do now it is much clearer what is being checked.

Reply:
So as I understand it, what this block of code is doing is checking for wind speed time points that either match the wind gust times, or are within wind gust time bounds (if for example the gust is a "max in hour").    So if the points aren't equal and the wind gust data is for a point time, rather than time period, the wind gust time doesn't have bounds and therefore the time coordinates don't match.  If there are bounds, the wind speed can match if it's for a validity time within the wind gust period.

So I think "Could not match time coordinate" is actually appropriate in both cases.

Reply:
Ok, fair enough